valgrind_memalign: delete sandbox options

yiqianwei opened the pull request:
valgrind does not support sandbox options,refer bz 1800495

ID:1869981

Signed-off-by: Yiqian Wei <yiwei@redhat.com>

jingzhao84 commented:
@yiqianwei  could you paste your test result here? thanks

yiqianwei commented:
@jingzhao84, OK , Please help me review this patch? thanks.

test result:
(1/1) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.valgrind_memalign.q35: PASS (21.33 s)

yiqianwei commented:
@jingzhao84, OK , Please help me review this patch? thanks.

test result:
(1/1) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.valgrind_memalign.q35: PASS (21.33 s)

yiqianwei commented:
@jingzhao84, OK , Please help me review this patch? thanks.
test result:
(1/1) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.valgrind_memalign.q35: PASS (21.33 s)

jingzhao84 commented:
Looks good on my side

@yanan-fu  could you help to review it?

Thanks
Jing

yanan-fu commented:
NACK

1. Please use correct format, follow:
https://avocado-framework.readthedocs.io/en/80.0/guides/contributor/chapters/styleguides.html

2. The problem is `valgrind  /usr/libexec/qemu-kvm ` does not support `sandbox` option, no matter it is `on` or `off`.
```
# valgrind  /usr/libexec/qemu-kvm -sandbox off
...
qemu-kvm: -sandbox off: There is no option group 'sandbox'

# valgrind  /usr/libexec/qemu-kvm -sandbox on
...
qemu-kvm: -sandbox on: There is no option group 'sandbox'
```
Here:
https://github.com/avocado-framework/avocado-vt/blob/728b1996cb6f8344c1dfa7db32f0ae442ad09b60/virttest/qemu_vm.py#L1457~1471

The solution should be `del qemu_sandbox`.

yiqianwei commented:
@yanan-fu,I have changed this patch,Can you help me review this patch again?

Test result:
(1/1) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.valgrind_memalign.q35: PASS (21.20 s)

yanan-fu commented:
> @yanan-fu,I have changed this patch,Can you help me review this patch again?
> 
> Test result:
> (1/1) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.valgrind_memalign.q35: PASS (21.20 s)

suggest:
```
# bz 1800495, valgrind don't support "sandbox" option
del qemu_sandbox
```
And explain the reason in commit info too.

yiqianwei commented:
@yanan-fu,I have added some content, please review me again, thank you

yiqianwei commented:
> > @yanan-fu,I have changed this patch,Can you help me review this patch again?
> > Test result:
> > (1/1) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.valgrind_memalign.q35: PASS (21.20 s)
> 
> suggest:
> 
> ```
> # bz 1800495, valgrind don't support "sandbox" option
> del qemu_sandbox
> ```
> 
> And explain the reason in commit info too.

@yanan-fu , I have added some content

yanan-fu commented:
@jingzhao84  call for your review, thanks!

jingzhao84 commented:
ACK by jinzhao@redhat.com
